html_css/v2/main-course/selectors.md

```diff
@@ -0,0 +1,264 @@
+### Introduction
+By now you should be comfortable with basic CSS selectors and have no trouble grabbing things by their type, class or ID. But to be a real CSS surgeon, sometimes you need more specialized tools. In this lesson we'll look at advanced CSS selectors and show you how to target elements in a more specific and finely grained way.
+
+These selectors can be especially useful when you can't (or don't want to) change your HTML markup.
+
+There are _a lot_ of advanced selectors, so going through every single one is outside the scope of this lesson. However, we'll go through some of the most useful and common ones, as well as arm you with the concepts and vocabulary to learn more on your own.
+
+As always feel free to open up your code editor and run you own experiments with these selectors - practice makes perfect!
+
+### Learning Outcomes
+* Understand how to use parent and sibling selectors
+* Recognize the difference between pseudo classes and pseudo elements
+* Learn about some of the most useful and common pseudo elements and pseudo classes
+* Learn about the different ways to select an attribute or its parts
+
+### Parent and Sibling Combinators
+Let's have a look at some more ways we can access different elements _without_ referring to their classes. Here are three new selectors to do just that.
+
+* `>` - the child combinator
+* `+` - the adjacent sibling combinator
+* `~` - the general sibling combinator
+
+We'll tackle some practical examples using this sample markup.
+
+~~~html
+<main class="parent">
+  <div class="child group1">
+    <div class="grand-child group1"></div>
+  </div>
+  <div class="child group2">
+    <div class="grand-child group2"></div>
+  </div>
+  <div class="child group3">
+    <div class="grand-child group3"></div>
+  </div>
+</main>
+~~~
+
+By now, you should be pretty comfortable writing rules using the descendant combinator you learned about in [foundations](https://www.theodinproject.com/paths/foundations/courses/foundations/lessons/css-foundations). For instance, if we wanted to select all the `child` and `grand-child` divs inside of `main`, we could write:
+
+~~~css
+main div {
+  /* Our cool CSS */
+}
+~~~
+
+But what if we wanted to be more specific and select _only_ the `child` or `grand-child` divs ? That's where the child combinator `>` comes in handy. Unlike the descendant combinator, it will only select direct children.
+
+~~~css
+/* This rule will only select divs with a class of child */
+main > div {
+  /* Our cool CSS */
+}
+
+/* This rule will only select divs with a class of grand-child */
+main > div > div {
+  /* More cool CSS */
+}
+~~~
+
+Phrased another way, the child selector will select an element that is one level of indentation down. In order to select an element that is adjacent to our target, or on the same level of indentation, we can use the adjacent sibling combinator `+`.
```

Comment:
Two proposed edits to line 61:

First, it currently says the child selector will select an element that is one level of indentation down. Unless I misunderstand, child selector will select *all* elements matched by the second selector, that are one level of indentation down. 

Second, I think that phonetically the second sentence on this line would flow better with the first comma removed. So it could be:

In order to select an element that is adjacent to our target or on the same level of indentation, we can use the adjacent sibling combinator `+`.

Reply:
In addition to this comment, I feel a little iffy about the indentation statement.. it's not _technically_ true, because you don't _have_ to use proper indentation in HTML.

I'd feel better about it if we had a little disclaimer.. something like:

...will select an element that is one level of indentation down (assuming you're indenting your code properly).

or

Phrased another way, if you have your code indented properly, the child selector will select an element that is one level of indentation down.

Reply:
I think that it is probably confusing for the two sentences to be together as a learner would be reading the first sentence which relates to the code above, while the second sentence refers to the code below.

I believe it'd be better to have the first sentence be put at line 47 or completely omitted to encourage learning the actual terminology.